test-context/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+use postgresql_embedded::{PostgreSQL, Settings, VersionReq};
+use std::collections::HashMap;
+use std::env;
+use std::env::current_dir;
+use std::io::ErrorKind;
+use std::ops::Index;
+use std::path::{Path, PathBuf};
+use test_context::AsyncTestContext;
+use tokio::io::AsyncReadExt;
+use tokio_util::bytes::Bytes;
+use tokio_util::io::ReaderStream;
+use tracing::{info_span, instrument, Instrument};
+use trustify_common as common;
+use trustify_module_ingestor::graph::Graph;
+use trustify_module_ingestor::model::IngestResult;
+use trustify_module_ingestor::service::{Format, IngestorService};
+use trustify_module_storage::service::fs::FileSystemBackend;
+
+#[allow(dead_code)]
+pub struct TrustifyContext {
+    pub db: common::db::Database,
+    pub graph: Graph,
+    pub storage: FileSystemBackend,
+    pub ingestor: IngestorService,
+    postgresql: Option<PostgreSQL>,
+}
+
+pub struct IngestionResults {
+    results: HashMap<String, Result<IngestResult, anyhow::Error>>,
+}
+
+impl Index<&str> for IngestionResults {
+    type Output = Result<IngestResult, anyhow::Error>;
+
+    #[allow(clippy::expect_used)]
+    fn index(&self, index: &str) -> &Self::Output {
+        self.results.get(index).expect("valid document path")
+    }
+}
+
+impl TrustifyContext {
+    pub async fn ingest_documents<'a, P: IntoIterator<Item = &'a str>>(
+        &self,
+        paths: P,
+    ) -> Result<IngestionResults, anyhow::Error> {
+        let workspace_root = find_workspace_root()?;
+        let test_data = workspace_root.join("etc").join("test-data");
+
+        let mut results = HashMap::new();
+
+        for path in paths {
+            results.insert(
+                path.to_string(),
+                self.ingest_document_inner(&test_data, path).await,
+            );
+        }
+
+        Ok(IngestionResults { results })
+    }
```

Comment:
I don't quite grok the need for introducing `IngestionResults`? I was expecting you'd return something like `paths.map(...).collect()` with a type of `Result<Vec<IngestResult>, _>` or some such.

Reply:
Mostly so we could index by path, instead of positional. I can alterate it though. I have no preference.

Reply:
Simplified as just a Vec<..> per your suggestion.